Feat(web-twig): Introduce Accordion for Twig #DS-448 (#606)
lmc-eu/spirit-design-system: packages/web-twig/src/Resources/components/Accordion/README.md

  id: 'AccordionExample',
} } %}
    {% block content %}
        {% embed "@spirit/accordionItem.twig" with { props: {
          id: `AccordionItemExample`,
        } } %}
            {% block content %}
                {% embed "@spirit/accordionHeader.twig" with { props: {
                  id: 'AccordionItemExampleHeader',
                  for: 'AccordionItemExampleContent',
                  slot: '<Link href="#">Link</Link><Pill color="selected">3</Pill>',
                  isOpen: true,
                } } %}
                    {% block content %}
                        Accordion Header
                    {% endblock %}
                {% endembed %}
                {% embed "@spirit/accordionContent.twig" with { props: {
                  id: 'AccordionItemExampleContent',
                  labelId: 'AccordionItemExampleHeader',
                  parent: 'AccordionExample',
                  isOpen: true,
                } } %}
                    {% block content %}
                        Accordion Content
                    {% endblock %}
                {% endembed %}
            {% endblock %}
        {% endembed %}
    {% endblock %}
{% endembed %}
```

## API

> The Accordion itself consists of many components which cannot be used independently.

Reviewer: The number is four, so I'd say `several` of `a few` rather than `many`… 🙂

Author: ok, fixed 🙂

### Accordion

| Prop name     | Type     | Default     | Required | Description              |
| ------------- | -------- | ----------- | -------- | ------------------------ |

> | `id`          | `string` | `null`      | no/yes   | Optional, depends on ... |

Reviewer: Depends on what? 🤔

Author: I didn't finish it anymore :/ ... fixed

| `class`       | `string` | `null`      | no       | Additional class name    |
| `elementType` | `string` | `'section'` | no       | Custom element type      |

### AccordionItem

| Prop name     | Type     | Default     | Required | Description           |
| ------------- | -------- | ----------- | -------- | --------------------- |
| `id`          | `string` | `null`      | yes      | Accordion Item ID     |
| `class`       | `string` | `null`      | no       | Additional class name |
| `elementType` | `string` | `'article'` | no       | Custom element type   |

### AccordionHeader

| Prop name     | Type     | Default | Required | Description              |
| ------------- | -------- | ------- | -------- | ------------------------ |
| `id`          | `string` | `null`  | yes      | AccordionHeader ID       |
| `class`       | `string` | `null`  | no       | Additional class name    |

> | `for`         | `string` | `null`  | yes      | Target element selector  |

Reviewer: We can specifiy the `for` prop is intended for `AccordionContent` ID.

Author: You're right, I finished writing it there

| `elementType` | `string` | `'h3'`  | no       | Custom element type      |
| `slot`        | `HTML`   | ``      | no       | Side slot in the header  |

> | `isOpen`      | `string` | `false` | no       | Whether is item selected |

Reviewer: ```suggestion
| `isOpen`      | `string` | `false` | no       | Whether the item is selected |
```

or even

```suggestion
| `isOpen`      | `string` | `false` | no       | Whether the item is expanded |
```

Author: fixed

### AccordionContent

| Prop name | Type     | Default | Required | Description                                                         |
| --------- | -------- | ------- | -------- | ------------------------------------------------------------------- |
| `id`      | `string` | `null`  | yes      | AccordionContent ID                                                 |
| `class`   | `string` | `null`  | no       | Additional class name                                               |
| `labelId` | `string` | `null`  | yes      | AccordionHeader ID                                                  |

> | `parent`  | `string` | `null`  | no       | A parent element selector that ensures that only one item is opened |

Reviewer: Maybe better:

```suggestion
| `parent`  | `string` | `null`  | no       | A parent element selector that ensures that only one item is open at a time |
```

Does it fit the line length limit though…?

Author: fixed

> | `isOpen`  | `string` | `false` | no       | Whether is item selected                                            |

Reviewer: Same as above, word order and `expanded` work better:

```suggestion
| `isOpen`  | `string` | `false` | no       | Whether the item is expanded                                            |
```

Author: fixed

    >
      Accordion Header
    </AccordionHeader>
    <AccordionContent id="AccordionItemExample3Content" labelledById="AccordionItemExample3Header">
      Accordion Content
    </AccordionContent>
  </AccordionItem>
  ...
</Accordion>
```

## Usage with initial open state (Stay open)

```html
<Accordion>
  <AccordionItem id="AccordionItemExample0">
    <AccordionHeader
      id="AccordionItemExample0Header"
      for="AccordionItemExample0Content"
      slot="<Link href='#'>Link</Link><Pill color='selected'>3</Pill>"
    >
      Accordion Header
    </AccordionHeader>
    <AccordionContent id="AccordionItemExample0Content" labelledById="AccordionItemExample0Header">
      Accordion Content
    </AccordionContent>
  </AccordionItem>
  <AccordionItem id="AccordionItemExample1">
    <AccordionHeader
      id="AccordionItemExample1Header"
      for="AccordionItemExample1Content"
      slot="<Pill color='selected'>3</Pill>"
      isOpen
    >
      Accordion Header
    </AccordionHeader>
    <AccordionContent id="AccordionItemExample1Content" labelledById="AccordionItemExample1Header" isOpen>
      Accordion Content
    </AccordionContent>
  </AccordionItem>
</Accordion>
```

> ## Usage with open only item one at a time

Reviewer: I believe this reads better 🙂 

```suggestion
## Usage with only one open item at a time
```

Author: fixed

## API

The Accordion itself consists of several components which cannot be used independently.

### Accordion

| Prop name     | Type     | Default     | Required | Description                                                                                           |
| ------------- | -------- | ----------- | -------- | ----------------------------------------------------------------------------------------------------- |

> | `id`          | `string` | `null`      | no/yes   | Optional, It depends on whether the "Always open" option is used, if so, the id field is not required |

Reviewer 2: I am suggesting some changes in copy.
- `Always open` -> `Stay open` - we are using it at the beginning of this readme
- `option` -> `functionality` - there is no "option" for this, I think that "functionality" better describes it
- `yes/no` -> `optional` - not sure about this, but seems better to me. Please decide yourself.

```suggestion
| `id`          | `string` | `null`      | optional   | It depends on whether the "Stay open" functionality is used. If so, the id field is not required. |
```

Author: ok, `optional` is also good for me, fixed

| `elementType` | `string` | `'section'` | no       | Custom element type                                                                                   |

### AccordionItem

| Prop name     | Type     | Default     | Required | Description           |
| ------------- | -------- | ----------- | -------- | --------------------- |
| `id`          | `string` | `null`      | yes      | AccordionItem ID      |
| `class`       | `string` | `null`      | no       | Additional class name |
| `elementType` | `string` | `'article'` | no       | Custom element type   |

### AccordionHeader

| Prop name     | Type     | Default | Required | Description                  |
| ------------- | -------- | ------- | -------- | ---------------------------- |
| `id`          | `string` | `null`  | yes      | AccordionHeader ID           |
| `class`       | `string` | `null`  | no       | Additional class name        |
| `for`         | `string` | `null`  | yes      | AccordionContent ID          |
| `elementType` | `string` | `'h3'`  | no       | Custom element type          |
| `slot`        | `HTML`   | ``      | no       | Side slot in the header      |

> | `isOpen`      | `string` | `false` | no       | Whether the item is expanded |

Reviewer 2: What about staying with the `open` status?

```suggestion
| `isOpen`      | `string` | `false` | no       | Whether the item is open |
```

Author: fixed

### AccordionContent

| Prop name | Type     | Default | Required | Description                                                                 |
| --------- | -------- | ------- | -------- | --------------------------------------------------------------------------- |
| `id`      | `string` | `null`  | yes      | AccordionContent ID                                                         |
| `class`   | `string` | `null`  | no       | Additional class name                                                       |
| `labelId` | `string` | `null`  | yes      | AccordionHeader ID                                                          |
| `parent`  | `string` | `null`  | no       | A parent element selector that ensures that only one item is open at a time |

> | `isOpen`  | `string` | `false` | no       | Whether the item is expanded                                                |

Reviewer 2: ```suggestion
| `isOpen`  | `string` | `false` | no       | Whether the item is open                                                |
```

Author: fixed
